cloud_provider/aws/aws_minion_manager.py

```diff
@@ -635,7 +656,28 @@ def check_scaling_group_instances(self, scaling_group):
                 # Wait for sometime before checking again.
                 time.sleep(60)
         return False
-
+    
+    def check_insufficient_capacity(self, scaling_group):
+        """
+        Checks whether not completed ASG activities got not have sufficient capacity error message.
+        """
+        # This error message from https://docs.aws.amazon.com/autoscaling/ec2/userguide/ts-as-capacity.html#ts-as-capacity-1
+        INSUFFICIENT_CAPACITY_MESSAGE = ['We currently do not have sufficient',
+                                           'capacity in the Availability Zone you requested']
+        
+        asg_info = scaling_group.get_asg_info()
+        response = AWSMinionManager.describe_asg_with_retries(
+            self._ac_client, asg_info.AutoScalingGroupName)
+        activities = response.Activities
```

Comment:
Same here: Not always is "Activities" is part of the response. This should change to:
`activities = response.Activities if "Activities" in response else {}`

Reply:
I just checked `describe-scaling-activities`. If no any activity for ASG, it will return an empty list.

```
$ aws autoscaling describe-scaling-activities --auto-scaling-group-name "new-dummy-asg"
{
  "Activities": []
}
```